src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -219,23 +226,30 @@ std::optional<Ball> SensorFusion::createBall(
     return std::nullopt;
 }
 
-
 Team SensorFusion::createFriendlyTeam(const std::vector<RobotDetection> &robot_detections)
 {
+    // TODO (1610) Implement friendly goalie ID override
+
     Team new_friendly_team =
         friendly_team_filter.getFilteredData(friendly_team, robot_detections);
-    RobotId friendly_goalie_id = sensor_fusion_config->FriendlyGoalieId()->value();
-    // TODO (1610) Implement friendly goalie ID override
     new_friendly_team.assignGoalie(friendly_goalie_id);
+    if (sensor_fusion_config->OverrideGameControllerFriendlyGoalieID()->value()) {
+        RobotId friendly_goalie_id_override = sensor_fusion_config->FriendlyGoalieId()->value();
+        new_friendly_team.assignGoalie(friendly_goalie_id_override);
```

Comment:
Should this be 
```suggestion
        new_friendly_team.assignGoalie(friendly_goalie_id);
```

Reply:
Line 235 is where I assign it friendly_goalie_id, then I check if its overridden and line 238 is where I re-assign it the overridden value if thats the case